src/data_gradients/feature_extractors/utils.py

```diff
@@ -1,40 +1,67 @@
+        :param prioritization_mode:    The prioritization_mode to get the top values for. One of:
+                - 'train_val_diff': Returns the top k rows with the biggest train_val_diff between 'train' and 'val' split values.
+                - 'outliers':       Returns the top k rows with the most extreme average values.
+                - 'max':            Returns the top k rows with the highest average values.
+                - 'min':            Returns the top k rows with the lowest average values.
+                - 'min_max':        Returns the (top k)/2 rows with the biggest average values, and the (top k)/2 with the smallest average values.
+        """
+        valid_modes = ("train_val_diff", "outliers", "max", "min", "min_max")
+        if prioritization_mode not in valid_modes:
+            raise ValueError(f"Invalid `prioritization_mode={prioritization_mode}'. Must be one of: {valid_modes}.")
+        self.topk = topk
+        self.prioritization_mode = prioritization_mode
 
-    aligned_train_histogram, aligned_val_histogram = {}, {}
-    for key in keys:
-        train_value = train_histogram.get(key)
-        val_value = val_histogram.get(key)
+    def select(self, df: pd.DataFrame, id_col: str, split_col: str, value_col: str):
+        """
+        Returns the top k rows of the DataFrame based on the prioritization_mode.
+        The DataFrame is expected to have at least three columns: id_col, split_col, val_col.
 
-        value_type = type(train_value) if train_value is not None else type(val_value)
-        default_value = value_type()
+        :param df:          The DataFrame to get the top values from.
+        :param id_col:      The name of the id column.
+        :param split_col:   The name of the split column. (Usually 'split')
+        :param value_col:   The name of column that will be used to calculate the metric.
+        :return: The Dataframe with only the rows associated to the most important values.
+        """
+        # Verify inputs
+        for col in [id_col, split_col, value_col]:
+            if col not in df.columns:
+                raise ValueError(f"{col} is not a column in the DataFrame")
 
-        aligned_train_histogram[key] = train_value or default_value
-        aligned_val_histogram[key] = val_value or default_value
+        # the mean of val_col for each id_col/split_col
+        df_mean = df.groupby([id_col, split_col])[value_col].mean().reset_index()
 
-    return aligned_train_histogram, aligned_val_histogram
+        # Pivot DataFrame to have 'train' and 'val' as columns
+        df_pivot = df_mean.pivot(index=id_col, columns=split_col, values=value_col)
 
+        # Calculate the relative difference or average based on the prioritization_mode
+        if self.prioritization_mode == "train_val_diff":
+            df_pivot["metric"] = np.abs((df_pivot["train"] - df_pivot["val"]) / ((df_pivot["train"] + df_pivot["val"]) / 2))
```

Comment:
1) Add a small `eps` in the denominator to avoid division by zero when train + val is 0 (It is technically possible I suppose). 
2) I don't quite like the fact we are using hard-coded split names here. What if we would have `test` or `holdout` split in addition? The code would either break or produce incorrect results. Is is possible to generalize this somehow by counting number of entries for each split with respect to total size?

Reply:
It is not completely generalizable because `train_val_diff` for instance assumes only 2 sets.
Here I pushed a change to work with any number of set for all the modesl, except for `train_val_diff`. For `train_val_diff` I make sure there are only 2 sets.

https://github.com/Deci-AI/data-gradients/pull/129/commits/84dac60b18e62b7f80366b949bd88e832ee97a2f

Reply:
epsilon added here [82207d3](https://github.com/Deci-AI/data-gradients/pull/129/commits/82207d36b1d35736c8b4f33821c9edce0e3e7413)